sources/web/datalab/polymer/components/datalab-editor/datalab-editor.ts

```diff
@@ -68,28 +64,34 @@ class DatalabEditorElement extends Polymer.Element {
       this._theme = settings.theme;
     }
 
-    // Get the file contents, or empty string if no path is specified
+    // Get the file contents, or empty string if no path is specified or the
+    // file could not be found.
     let content = '';
-    if (this.filePath) {
+    if (this.fileId) {
       this._busy = true;
-      // Passing the asText=true parameter guarantees the returned type is not a directory.
-      // An error is thrown if it is.
-      this._file = await this._fileManager.get(this.filePath, true /*asText*/)
-        .catch((e: Error) => {
-          Utils.showErrorDialog('Error', e.message);
-          return null;
-        });
+      try {
+        this._fileManager = FileManagerFactory.getInstanceForType(this.fileId.source);
+
+        // Get the file object and its contents
+        this._file = await this._fileManager.get(this.fileId);
+        content =
+            (await this._fileManager.getContent(this.fileId, true /*asText*/)).getEditorText();
+      } catch (e) {
+        Utils.showErrorDialog('Error loading file', e.message);
+        this.fileId = null;
+      }
 
       this._busy = false;
-
-      if (this._file) {
-        content = this._file.content as string;
-      }
+    } else {
+      // TODO: Make this more flexible instead of assuming the default
+      // destination is jupyter. If there's no fileId specified (blank editor),
+      // we should ask the user on save about the file destination.
+      this._fileManager = FileManagerFactory.getInstanceForType(FileManagerType.JUPYTER);
```

Comment:
Something we can consider for a future PR is to implement some kind of "null" FileManager that is not tied to anything. Then when the user asks to Save, we can figure out where the user wants to save, and replace this._fileManager with the appropriate FileManager for that destination.